yarn-project/foundation/src/json-rpc/convert.ts

```diff
@@ -1,6 +1,25 @@
 import { ClassConverter } from './class_converter.js';
 import { Buffer } from 'buffer';
 
+/**
+ * Recursively looks through an object for bigints and converts them to object format.
+ * @param obj - The object to convert.
+ * @returns The converted object with stringified bigints.
+ */
+export const convertBigintsInObj = (obj: any) => {
+  for (const i in obj) {
+    if (typeof obj[i] === 'bigint') {
+      obj[i] = {
+        type: 'bigint',
+        data: obj[i].toString(),
```

Comment:
Do we need to do this in a separate recursive pass?

Reply:
sorry I missed this in the off-site.
We need it for structures that have nested `bigint`s. I was using `JsonStringify` previously but because the koa body parser already does that (but doesn't let you change how in order to fix it for bigints) we currently have doubly-stringified JSON objects which is kinda ugly + adds some unnecessary size to the response